### ISSUE TITLE
Refinements for some ("global") classes/modules not working

### ISSUE THREAD

**Issue by palkan** [jruby/jruby#5220]
### Environment

* JRuby version: jruby 9.2.0.0 (2.5.0) 2018-05-24 81156a8 OpenJDK 64-Bit Server VM 25.171-b11 on 1.8.0_171-8u171-b11-1~deb9u1-b11 +jit [linux-x86_64]
* Operating system and platform: Linux 3971953e9581 4.9.49-moby #1 SMP Fri Dec 8 13:40:02 UTC 2017 x86_64 GNU/Linux

### Expected Behavior

The following refinements are expected to work (as they are in CRuby):

```ruby
module ObjectExt
  # the same with BasicObject and Kernel 
  refine Object do
    def i_am_refined
      'i_am_refined'
    end
  end
end

using ObjectExt

class Test; end

p Test.new.i_am_refined
```

### Actual Behavior

```
NoMethodError: undefined method `i_am_refined' for #<Test:0x2d8f65a4>
```

Other _non-refineables_:
- `Module`
- `Numeric`:

This doesn't work:

```ruby
module NumericExt
  refine Numeric do
    def i_am_refined
      'i_am_refined'
    end
  end
end

using NumericExt

1.i_am_refined
```

Although this works:

```ruby
module NumericExt
  refine Integer do
    def i_am_refined
      'i_am_refined'
    end
  end
end

using NumericExt

1.i_am_refined
```

**Comment by palkan** [comment]
Probably a duplicate of https://github.com/jruby/jruby/issues/4330.

And the `Numeric` example looks like https://github.com/jruby/jruby/issues/4288 (but it's closed).

**Comment by headius** [comment]
Fixed in 9.2.7 with the refinements work.
